src/factories/calorimetry/CalorimeterHitReco_factory.h

```diff
@@ -28,7 +28,7 @@ class CalorimeterHitReco_factory : public JOmniFactory<CalorimeterHitReco_factor
     ParameterRef<double> m_resolutionTDC {this, "resolutionTDC", config().resolutionTDC};
     ParameterRef<double> m_thresholdFactor {this, "thresholdFactor", config().thresholdFactor};
     ParameterRef<double> m_thresholdValue {this, "thresholdValue", config().thresholdValue};
-    ParameterRef<double> m_samplingFraction {this, "samplingFraction", config().sampFrac};
+    ParameterRef<std::string> m_samplingFraction {this, "samplingFraction", config().sampFrac};
```

Comment:
Though this probably doesn't handle variants... Maybe we can somehow 'pick one' here, naturally the string. Maybe:
```
    ParameterRef<std::string> m_samplingFraction {this, "samplingFraction", std::get<std::string>(config().sampFrac)};
```